pkg/resourcemapping/orm_client_manager.go

```diff
@@ -69,69 +70,55 @@ type OwnerResources struct {
+
+	// Find ORM v2 given the source owned resource
+	var ownedResourcePathObjs []*v1alpha1.ResourcePath
+	for _, ownedResourcePath := range ownedResourcePaths {
+		var ownedResourcePathObj = &v1alpha1.ResourcePath{
 			ObjectReference: owned,
-			Path:            path,
+			Path:            ownedResourcePath,
 		}
-		sourceResourcePath = append(sourceResourcePath, resourcePath)
+		ownedResourcePathObjs = append(ownedResourcePathObjs, ownedResourcePathObj)
 		// If there are nested/hierarchy of owners, this always fetch the top owner resource. any updates on the top
-		// owner resource will flow down the chain to get it's children to update
-		ownerResourcePaths := manager.SeekTopOwnersResourcePathsForOwnedResourcePath(*resourcePath)
-		// ownerResourcePaths will never be empty, as it returns the source/owned resource kind if it cannot
-		// find the corresponding owner resources and it's paths. so check if the ownerResourcePaths returned
-		// are same as resourcePath, if same then it didn't find any mappings related to V2 orm
-		for _, ownerResourcePath := range ownerResourcePaths {
-			if ownerResourcePath != *resourcePath {
-				allOwnerResourcePaths[path] = ownerResourcePaths
+		// owner resource will flow down the chain to get its children to update.
+		ownerResourcePathObjs := ormClientMgr.SeekTopOwnersResourcePathsForOwnedResourcePath(*ownedResourcePathObj)
+		// The ownerResourcePaths will never be empty, as the SeekTopOwnersResourcePathsForOwnedResourcePath function
+		// returns the owned resource kind if it cannot find the corresponding owner resources and their paths. So
+		// check if the ownerResourcePaths returned are the same as resourcePath, if same then it didn't find any
+		// mappings related to V2 ORM.
+		for _, ownerResourcePath := range ownerResourcePathObjs {
+			if ownerResourcePath != *ownedResourcePathObj {
+				allOwnerResourcePathObjs[ownedResourcePath] = ownerResourcePathObjs
 			}
 		}
 	}
 
-	if len(allOwnerResourcePaths) > 0 {
+	foundORMV1 := false
+	if len(allOwnerResourcePathObjs) > 0 {
 		glog.Infof("Found owner resource paths using ORM v2 for owned object %s:%s:%s",
 			ownedObj.GetKind(), ownedObj.GetNamespace(), ownedObj.GetName())
-	}
-	// find legacy orm if cannot locate v2 orm
-	if len(allOwnerResourcePaths) == 0 {
-		allOwnerResourcePaths, err = manager.LocateOwnerPaths(ownedObj, ownerReference, sourceResourcePath)
+	} else {
+		// Find legacy orm v1 if orm v2 is empty
+		glog.Infof("Cannot locate owner resource paths using ORM v2 for owned object %s:%s:%s. Try ORM v1 ...",
+			ownedObj.GetKind(), ownedObj.GetNamespace(), ownedObj.GetName())
+		allOwnerResourcePathObjs, err = ormClientMgr.LocateOwnerPaths(ownedObj, ownerReference, ownedResourcePathObjs)
 		if err != nil {
-			return &OwnerResources{
-				ControllerObj:     ownedObj,
-				OwnerResourcesMap: allOwnerResourcePaths,
-				isV1ORM:           foundORMV1,
-			}, err
+			return nil, err
```

Comment:
In keeping it consistent with the v2 orm client, it was decided to return source path if there is any sort of error while fetching the v1 orms.

Reply:
In the original implementation, when there is an error, the `OwnerResources` is ignored. So there is no need to construct this object.